var/spack/repos/builtin/packages/occa/package.py

```diff
@@ -33,91 +38,177 @@ class Occa(Package):
-        # disabled. Otherwise, let OCCA autodetect what is available.
-
+        # Following only apply to the make build. For each variant, set the
+        # environment variable OCCA_{CUDA,HIP,OPENMP}_ENABLED only if the
+        # variant is enabled. We disable OCCA autodetect as it causes issues
+        # on head nodes.
         if "+cuda" in spec:
             cuda_dir = spec["cuda"].prefix
             cuda_libs_list = ["libcuda", "libcudart", "libOpenCL"]
             cuda_libs = find_libraries(cuda_libs_list, cuda_dir, shared=True, recursive=True)
             env.set("OCCA_INCLUDE_PATH", cuda_dir.include)
             env.set("OCCA_LIBRARY_PATH", ":".join(cuda_libs.directories))
+            env.set("OCCA_CUDA_ENABLED", "1")
         else:
             env.set("OCCA_CUDA_ENABLED", "0")
 
         # Disable hip autodetection for now since it fails on some machines.
-        env.set("OCCA_HIP_ENABLED", "0")
+        if "+rocm" in spec:
+            hip_dir = spec["hip"].prefix
+            hip_libs_list = ["amdhip64"]
+            hip_libs = find_libraries(hip_libs_list, hip_dir, shared=True, recursive=True)
+            env.set("OCCA_INCLUDE_PATH", hip_dir.include)
+            env.set("OCCA_LIBRARY_PATH", ":".join(hip_libs.directories))
+            env.set("OCCA_HIP_ENABLED", "1")
+        else:
+            env.set("OCCA_HIP_ENABLED", "0")
 
-        if "~opencl" in spec:
-            env.set("OCCA_OPENCL_ENABLED", "0")
+        # Turn-off OpenCL since it fails on some of the machines.
+        env.set("OCCA_OPENCL_ENABLED", "0")
 
-        if "~openmp" in spec:
+        if "+openmp" in spec:
+            env.set("OCCA_OPENMP_ENABLED", "1")
+        else:
             env.set("OCCA_OPENMP_ENABLED", "0")
 
         # Setup run-time environment for testing.
         env.set("OCCA_VERBOSE", "1")
-        self._setup_runtime_flags(env)
-
-    def setup_run_environment(self, env):
-        # The 'env' is included in the Spack generated module files.
-        self._setup_runtime_flags(env)
+        self.setup_run_environment(env)
 
     def setup_dependent_build_environment(self, env, dependent_spec):
         # Export OCCA_* variables for everyone using this package from within
         # Spack.
-        self._setup_runtime_flags(env)
+        self.setup_run_environment(env)
+
+
+class CMakeBuilder(CMakeBuilder, SetupEnvironment):
+    def cmake_args(self):
+        spec = self.spec
+        args, include_dirs, library_dirs = [], [], []
+
+        if "+cuda" in spec:
+            cuda_dir = spec["cuda"].prefix
+            cuda_libs_list = ["libcuda", "libcudart", "libOpenCL"]
+            cuda_libs = find_libraries(cuda_libs_list, cuda_dir, shared=True, recursive=True)
+            include_dirs.append(cuda_dir.include)
+            library_dirs.extend(cuda_libs.directories)
+            args.append("-DOCCA_ENABLE_CUDA=ON")
+        else:
+            args.append("-DOCCA_ENABLE_CUDA=OFF")
```

Comment:
Here we can use `self.define_from_variant` We can also use `self.define` below instead of raw strings.

Reply:
Thanks ! Fixed in the latest commit.